x-pack/plugins/observability_solution/apm/server/assistant_functions/get_apm_dataset_info.ts

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { FunctionVisibility } from '@kbn/observability-ai-assistant-plugin/common';
+import { compact, mapValues, omit, uniq } from 'lodash';
+import datemath from '@elastic/datemath';
+import { rangeQuery } from '@kbn/observability-plugin/server';
+import type { FunctionRegistrationParameters } from '.';
+
+export function registerGetApmDatasetInfoFunction({
+  apmEventClient,
+  registerFunction,
+  resources,
+}: FunctionRegistrationParameters) {
+  registerFunction(
+    {
+      name: 'get_apm_dataset_info',
+      contexts: ['core'],
+      visibility: FunctionVisibility.AssistantOnly,
+      description: `Use this function to get information about APM data.`,
+      parameters: {
+        type: 'object',
+        additionalProperties: false,
+        properties: {
+          start: {
+            type: 'string',
+            description:
+              'The start of the current time range, in datemath, like now-24h or an ISO timestamp',
+          },
+          end: {
+            type: 'string',
+            description:
+              'The end of the current time range, in datemath, like now-24h or an ISO timestamp',
+          },
+        },
+      } as const,
+    },
+    async (
+      { arguments: { start, end } },
+      signal
+    ): Promise<{
+      content: {
+        fields: string[];
+        description: string;
+      };
+    }> => {
+      const indices = omit(apmEventClient.indices, 'sourcemap', 'onboarding');
+
+      const allIndices = uniq(
+        Object.values(indices)
+          .flatMap((indicesForEvent) => indicesForEvent.split(','))
+          .map((index) => index.trim())
+      );
+
+      const esClient = (await resources.context.core).elasticsearch.client
+        .asCurrentUser;
```

Comment:
nit: wdyt about using the [typed search client](https://github.com/elastic/kibana/blob/9195b22667fcd81eacd97baaa33becb062a04fc6/x-pack/plugins/observability_solution/apm/server/routes/diagnostics/create_typed_es_client.ts#L17-L26). You might want to move it out of diagnostics if you do.

```suggestion
      const esClient = (await resources.context.core).elasticsearch.client
        .asCurrentUser;
        
        
```

Reply:
That's fair, I wasn't using it before because I was also calling fieldCaps, but I'm no longer using it, so will change to the typed client.